### ISSUE TITLE
result

### ISSUE THREAD

**Issue by zhangxuan-thecastle** [sotiraslab/AgileFormer#17]
Hello author, may I ask why I cannot achieve 85.74 on the synapse dataset with the same environment and parameters using 3090

**Comment by peijieq** [comment]
Hi @zhangxuan-thecastle, I think there is a mistake in the previous learning rate. The learning rate that was used for training the 2D Synapse multi-organ segmentation was 2e-4. Sorry for the confusion (I found this error after revisiting the code and checkpoints). Could you try this new learning rate  and say if you can you reproduce the results. Another possible reason is that the random seed might not be perfectly fixed in some system. You may also run a few trails to verify this.

**Comment by zhangxuan-thecastle** [comment]
> Hi @zhangxuan-thecastle, I think there is a mistake in the previous learning rate. The learning rate that was used for training the 2D Synapse multi-organ segmentation was 2e-2. Sorry for the confusion (I found this error after revisiting the code and checkpoints). Could you try this new learning rate and say if you can you reproduce the results. Another possible reason is that the random seed might not be perfectly fixed in some system. You may also run a few trails to verify this.

Hello author, could you please provide me with the configurations for agileFormer_mase.yaml and train.by for the 2Dsynapse dataset and ACDC dataset? Your article is really well written, and I am eager to reproduce a good result. Thank you very much

**Comment by peijieq** [comment]
You do not need to change the config file and the train.py file. But you need to add the learning rate argument to the training script, i.e., ```python train.py --base_lr 2e-4```

**Comment by zhangxuan-thecastle** [comment]
> Hi @zhangxuan-thecastle, I think there is a mistake in the previous learning rate. The learning rate that was used for training the 2D Synapse multi-organ segmentation was 2e-4. Sorry for the confusion (I found this error after revisiting the code and checkpoints). Could you try this new learning rate and say if you can you reproduce the results. Another possible reason is that the random seed might not be perfectly fixed in some system. You may also run a few trails to verify this.

Hello author, could you please provide me with the configurations for agileFormer_mase.yaml and train.exe for the 2Dsynapse dataset and ACDC dataset? Your article is really well written, and I am eager to reproduce a good result. Thank you very much

**Comment by zhangxuan-thecastle** [comment]
> You do not need to change the config file and the train.py file. But you need to add the learning rate argument to the training script, i.e., `python train.py --base_lr 2e-4`

This is my train.exe configuration
parser.add_argument('--max_epochs', type=int,
                    default=400, help='maximum epoch number to train')
parser.add_argument('--dice_loss_weight', type=float,
                    default=0.6, help='loss balance factor for th dice loss')#0.6
parser.add_argument('--batch_size', type=int,
                    default=24, help='batch_size per gpu')#24
parser.add_argument('--n_gpu', type=int, default=1, help='total gpu')
parser.add_argument('--deterministic', type=int,  default=1,
                    help='whether use deterministic training')
parser.add_argument('--optimizer', type=str,  default='AdamW',
                    help='the choice of optimizer')
parser.add_argument('--base_lr', type=float,  default=2e-4,
                    help='segmentation network learning rate')
parser.add_argument('--weight_decay', type=float,  default=1e-4,
                    help='weight decay')
parser.add_argument('--clip_grad', type=float,  default=8,
                    help='gradient norm')
parser.add_argument('--lr_scheduler', type=str,  default='cosine',
                    help='the choice of learning rate scheduler')
parser.add_argument('--warmup_epochs', type=int,
                    default=20, help='learning rate warm up epochs')
parser.add_argument('--seed', type=int,
                    default=2222, help='random seed')
parser.add_argument('--cfg', type=str, required=True,   
                    metavar="FILE", help='path to config file', )
parser.add_argument('--resume', help='resume from checkpoint')

This is my agileformer_mase.yaml configuration
Params:
    img_size: 224
    patch_size: 4
    deep_supervision: True
    expansion: 4
    dim_stem: 128
    dims: [128, 256, 512, 1024]
    depths_Encoder: [2, 4, 10, 2]
    depths_Decoder: [2, 4, 10, -1]
    stage_spec: [[N, D], [N, D, N, D], [N, D, N, D, N, D, N, D, N, D], [D, D]]
    heads: [4, 8, 16, 32]
    window_sizes: [7, 7, 7, 7]
    deform_dec: [False, False, False, False]
    groups: [2, 4, 8, 16]
    encoder_pos_layers: [-1]
    decoder_pos_layers: [1, 2, 3]
    use_pes: [False, False, False, False]
    dwc_pes: [False, False, False, False]
    strides: [8, 4, 2, 1]
    offset_range_factor: [-1, -1, -1, -1]
    no_offs: [False, False, False, False]
    fixed_pes: [False, False, False, False]
    use_dwc_mlps: [True, True, True, True]
    use_lpus: [True, True, True, True]
    use_conv_patches: True
    ksizes: [9, 7, 5, 3]
    nat_ksizes: [7, 7, 7, 7]
    drop_rate: 0.0
    attn_drop_rate: 0.0
    drop_path_rate: 0.7  
    
    Do you see anything wrong? I can only get around 84.7 on a 3090 graphics card with this configuration

**Comment by peijieq** [comment]
All the configuration seems correct to me. Let me double-check it and come back to you (It has been a while, I may take some time for me to test out what is the possible reason for the performance degradation on your end). In the mean time, could you try to run the experiment multiple times to see if the same result is obtained.

**Comment by zhangxuan-thecastle** [comment]
> All the configuration seems correct to me. Let me double-check it and come back to you (It has been a while, I may take some time for me to test out what is the possible reason for the performance degradation on your end). In the mean time, could you try to run the experiment multiple times to see if the same result is obtained.

Yes, I have run this configuration many times and it is around 84.7. When base_1r is set to 3e-4, it can reach around 84.9

**Comment by peijieq** [comment]
Thanks for your efforts, I will thoroughly investigate the code.

**Comment by zhangxuan-thecastle** [comment]
> Thanks for your efforts, I will thoroughly investigate the code.

In addition, when I use tvdcn for deformable convolution, the effect is only about 84.5, while Pytorch provides a deformable convolution effect of about 84.7.I kindly request the author to verify and send me the train.py and agileformer_base.yaml configurations with an effect of around 85.7. My email is zhangxuanxuan@s.dlu.edu.cn Thank you.

**Comment by zhangxuan-thecastle** [comment]
> Thanks for your efforts, I will thoroughly investigate the code.

Hello author, may I ask if you have obtained good results

**Comment by zhangxuan-thecastle** [comment]
> Thanks for your efforts, I will thoroughly investigate the code.

Hello author, may I ask if you have obtained good results

**Comment by zhangxuan-thecastle** [comment]
> Thanks for your efforts, I will thoroughly investigate the code.

Hello author, have you obtained good results? I am eager to use your code for improvement

**Comment by peijieq** [comment]
Sorry for the delay response and thank you for your patience. I understand your urgency, but I am afraid I cannot (though I want to) run the experiments at this moment due to a few very tight deadlines I have to meet. Instead, I will upload the segmentation results shortly in case you want to use it for visualization.

**Comment by zhangxuan-thecastle** [comment]
> Sorry for the delay response and thank you for your patience. I understand your urgency, but I am afraid I cannot (though I want to) run the experiments at this moment due to a few very tight deadlines I have to meet. Instead, I will upload the segmentation results shortly in case you want to use it for visualization.

Okay, I look forward to you uploading it as soon as possible. Your paper is well written and I really want to improve it based on a good result, and then cite your paper

**Comment by zhangxuan-thecastle** [comment]
> Sorry for the delay response and thank you for your patience. I understand your urgency, but I am afraid I cannot (though I want to) run the experiments at this moment due to a few very tight deadlines I have to meet. Instead, I will upload the segmentation results shortly in case you want to use it for visualization.

Hello author, may I ask if you have achieved good results? I am really anxious to find parameters for a good result

**Comment by LorenzoSincere** [comment]
> > All the configuration seems correct to me. Let me double-check it and come back to you (It has been a while, I may take some time for me to test out what is the possible reason for the performance degradation on your end). In the mean time, could you try to run the experiment multiple times to see if the same result is obtained.
> 
> Yes, I have run this configuration many times and it is around 84.7. When base_1r is set to 3e-4, it can reach around 84.9

Hello, may I ask what is the version of ptorch you are running and if you have encountered any situations where tvdcn cannot be used. My pytorch 2.2 cannot be used with tvdcn, 2.1 can be used. thank you. I reproduced the Tiny version W DS, which differs from the paper by 0.849 on V100

**Comment by zhangxuan-thecastle** [comment]
> Hello, may I ask what is the version of ptorch you are running and if you have encountered any situations where tvdcn cannot be used. My pytorch 2.2 cannot be used with tvdcn, 2.1 can be used. thank you. I reproduced the Tiny version W DS, which differs from the paper by 0.849 on V100

Yes, I am unable to use TVDCN. There is also a discrepancy between my reproduced results and those in the paper
